src/Microsoft.ApplicationInsights/Metrics/Implementation/ConcurrentDatastructures/MultidimensionalCube2.cs

```diff
@@ -342,6 +370,11 @@ private MultidimensionalPointResult<TPoint> TryCreatePoint(string[] coordinates,
             }
 
             { 
+                if (appliedUnsafeDimensionValuesCapFallback)
+                {
+                    pointMoniker = BuildPointMoniker(coordinates);
```

Comment:
this should be done outside of the createpoint call, as "other' may have already created.

Reply:
Good catch. However, not outside, but a little further up, while you are under the same lock. PR updated.